add_to_app/android_view/app/src/main/java/dev/flutter/example/androidView/FlutterViewEngine.kt

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package dev.flutter.example.androidView
+
+import android.content.Intent
+import android.util.Log
+import androidx.activity.ComponentActivity
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import io.flutter.embedding.android.FlutterView
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.plugin.platform.PlatformPlugin
+
+/**
+ *  This is an application specific wrapper class that exists to expose the intersection of an
+ *  active activity and a visible view to a FlutterEngine.
+ *
+ *  <p>Omitted features from the FlutterActivity:
+ *  <ul>
+ *  <li>State restoration. If you're integrating at the view level, you should handle activity
+ *      state restoration yourself.
+ *  <li>Engine creations. At this level of granularity, you must make an engine and attach.
+ *      and all engine features like initial route etc must be configured on the engine yourself.
+ *  <li>Splash screens. You must implement it yourself. Read from addOnFirstFrameRenderedListener
+ *      as needed.
+ *  <li>Transparency, surface/texture. These are just FlutterView level APIs. Set them on the
+ *      FlutterView directly.
+ *  <li>Intents. This doesn't do any translation of intents into actions in the FlutterEngine.
+ *      you must do them yourself.
+ *  <li>Back buttons. You must decide whether to send it to Flutter via
+ *      FlutterEngine.getNavigationChannel.popRoute(), or consume it natively. Though that decision
+ *      may be difficult due to https://github.com/flutter/flutter/issues/67011.
+ *  </ul>
+ *
+ * Your own FlutterView integrating application may need a similar wrapper but
+ * you must decide when the FlutterView is active based on your own application's
+ * intent.
+ */
+class FlutterViewEngine(val engine: FlutterEngine) : LifecycleObserver{
+    private var flutterView: FlutterView? = null
+    private var activity: ComponentActivity? = null
+    private var platformPlugin: PlatformPlugin? = null
+
+    // This is the intersection of an available activity and of a visible
+    // FlutterView. This is where Flutter would start rendering.
```

Comment:
Is the idiomatic docstring format for kotlin?  This docstring doesn't exactly say what this method does.  It more describes what it is.

Reply:
it's a private method. But for consistency, I made everything use the kdoc format (which provides nice highlighting etc)